flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/operators/route/RouteFunction.java

```diff
@@ -91,14 +107,36 @@ public Event map(Event event) throws Exception {
                         event.getClass().getCanonicalName()));
         ChangeEvent changeEvent = (ChangeEvent) event;
         TableId tableId = changeEvent.tableId();
-
+        // priority table route
         for (Tuple2<Selectors, TableId> route : routes) {
             Selectors selectors = route.f0;
             TableId replaceBy = route.f1;
             if (selectors.isMatch(tableId)) {
                 return recreateChangeEvent(changeEvent, replaceBy);
             }
         }
+
+        // then schema route
+        for (Tuple4<String, String, String, String> schemaRoute : schemaRoutes) {
+            if (schemaRoute.f0.equals(tableId.getSchemaName())) {
+                // database mapping, table add prefix and suffix mapping
+                String replaceIdentifier;
+                String replaceTableName =
+                        Optional.ofNullable(schemaRoute.f2).orElse("")
+                                + tableId.getTableName()
+                                + Optional.ofNullable(schemaRoute.f3).orElse("");
+                if (tableId.getNamespace() != null) {
+                    replaceIdentifier =
+                            String.join(
+                                    ".", tableId.getNamespace(), schemaRoute.f1, replaceTableName);
+                } else {
+                    replaceIdentifier = String.join(".", schemaRoute.f1, replaceTableName);
+                }
+
+                return recreateChangeEvent(changeEvent, TableId.parse(replaceIdentifier));
```

Comment:
We can use `new TableId(tableId.getSchemaName(), schemaRoute.f1, replaceTableName)` to simplify the `join` and `parse` code.

Reply:
get. i will process it，but there only hava TableId.tableId() function can be use，it is ok?
```
// then schema route
        for (Tuple4<String, String, String, String> schemaRoute : schemaRoutes) {
            if (schemaRoute.f0.equals(tableId.getSchemaName())) {
                // database routing, table add prefix and suffix mapping
                TableId replaceBy;
                String replaceTableName =
                        Optional.ofNullable(schemaRoute.f2).orElse("")
                                + tableId.getTableName()
                                + Optional.ofNullable(schemaRoute.f3).orElse("");
                if (tableId.getNamespace() != null) {
                    replaceBy = TableId.tableId(tableId.getNamespace(),schemaRoute.f1, replaceTableName);
                } else {
                    replaceBy = TableId.tableId(schemaRoute.f1, replaceTableName);
                }

                return recreateChangeEvent(changeEvent, replaceBy);
            }
        }
```

Reply:
> but there only hava TableId.tableId() function can be use，it is ok?
Yeah, This will be more concise.